### PR TITLE
Adding of skeleton for master-gateway

### PR DESCRIPTION
**Pull request by Ma-Shell**
Added basic skeleton for master-gateway and restructured most parts

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,418 +1,27 @@
...
-	keys = make(map[string]rsa.PrivateKey)
-	readKeys()
-	//log.Println(keys)
-
-	// Connect to rabbitmq
-	connectRabbit()
-
-	// Setup the HTTP-listener
-	initHTTP()
 }
```
Comment (webstergd) [inline main.go]: Please add a new line

File: mastergateway/mastergateway.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	tasking.FailOnError(err, "Couldn't read config file")
+	
+	// Parse the public keys
+	keys = make(map[string]*rsa.PublicKey)
+	readKeys()
+	//log.Println(keys)
+
+	// Setup the HTTP-listener
+	initHTTP()
+}
```
Comment (webstergd) [inline mastergateway/mastergateway.go]: New line

File: utils/tasking.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	label := []byte("")
+	ciphertext, err := rsa.EncryptOAEP(sha256.New(), rand.Reader, key, plaintext, label)
+	return ciphertext, err
+}
+
+func RsaDecrypt(ciphertext []byte, key *rsa.PrivateKey) ([]byte, error) {
+	label := []byte("")
+	plaintext, err := rsa.DecryptOAEP(sha256.New(), rand.Reader, key, ciphertext, label)
+	return plaintext, err
+}
```
Comment (webstergd) [inline utils/tasking.go]: \n

File: gateway/gateway.go [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+	keys = make(map[string]*rsa.PrivateKey)
+	readKeys()
+	//log.Println(keys)
+
+	// Connect to rabbitmq
+	connectRabbit()
+
+	// Setup the HTTP-listener
+	initHTTP()
+}
```
Comment (webstergd) [inline gateway/gateway.go]: please add a new line

**Comment by Ma-Shell** [comment]
newlines have been added as requested

**Pull request merged**